### PR TITLE
Fixes cooking value not updating properly

### LINKED ISSUE #6602: Cook point save only last cook point
* **rAthena Hash**:  a1b7fab19ba263bb29eb4d10e55fe81868b4850c

* **Client Date**: 20200401

* **Server Mode**:  Pre-renewal

* **Description of Issue**: Cook 10 recipes
  * Result: You only got the points from last cooked recipe when you log off.
  * Expected Result: Points should be calculated and stored in sd->cook_mastery at each cooking

Possible resolution :

/src/map/chrif.cpp
Function : int chrif_save(struct map_session_data *sd, int flag)
after chrif_bsdata_save(sd, ((flag&CSAVE_QUITTING) && !(flag&CSAVE_AUTOTRADE)));
add
pc_setglobalreg(sd, add_str(COOKMASTERY_VAR), sd->cook_mastery);

src/map/skill.cpp
Function : bool skill_produce_mix(struct map_session_data *sd, uint16 skill_id, t_itemid na
Replace
pc_setglobalreg(sd, add_str(COOKMASTERY_VAR), sd->cook_mastery + ( 1 << ( (skill_produce_db[idx].itemlv - 11) / 2 ) ) * 5);
by
sd->cook_mastery = sd->cook_mastery + (( 1 << ( (skill_produce_db[idx].itemlv - 11) / 2 ) ) * 5);

Replace
pc_setglobalreg(sd, add_str(COOKMASTERY_VAR), sd->cook_mastery - ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) - ( ( ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) >> 1 ) * 3 ));
by
sd->cook_mastery = sd->cook_mastery - ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) - ( ( ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) >> 1 ) * 3 );

### PR DESCRIPTION
**Pull request by aleos89**
* **Addressed Issue(s)**: Fixes #6602

* **Server Mode**: Pre-renewal and Renewal

* **Description of Pull Request**: 
  * Resolves cooking items only updating the player's cooking variable and not properly updating the player's session data along with it.
Thanks to @Shaktohh and @Lemongrass3110!

### CODE REVIEW

File: src/map/skill.cpp [hunk 1]
```diff
@@ -21833,7 +21833,7 @@ bool skill_produce_mix(struct map_session_data *sd, uint16 skill_id, t_itemid na
 					if (skill_produce_db[idx].itemlv > 10 && skill_produce_db[idx].itemlv <= 20) { //Cooking items.
 						clif_specialeffect(&sd->bl, EF_COOKING_OK, AREA);
 						if (sd->cook_mastery < 1999)
```
Comment (Lemongrass3110) [inline src/map/skill.cpp:21835]:
```suggestion
```

File: src/map/skill.cpp [hunk 2]
```diff
@@ -21833,7 +21833,7 @@ bool skill_produce_mix(struct map_session_data *sd, uint16 skill_id, t_itemid na
...
-							pc_setglobalreg(sd, add_str(COOKMASTERY_VAR), sd->cook_mastery + ( 1 << ( (skill_produce_db[idx].itemlv - 11) / 2 ) ) * 5);
+							pc_setparam(sd, SP_COOKMASTERY, sd->cook_mastery + ( 1 << ( (skill_produce_db[idx].itemlv - 11) / 2 ) ) * 5);
```
Comment (Lemongrass3110) [inline src/map/skill.cpp:21836]:
```suggestion
						pc_setparam(sd, SP_COOKMASTERY, sd->cook_mastery + ( 1 << ( (skill_produce_db[idx].itemlv - 11) / 2 ) ) * 5);
```

File: src/map/skill.cpp [hunk 3]
```diff
@@ -21988,7 +21988,7 @@ bool skill_produce_mix(struct map_session_data *sd, uint16 skill_id, t_itemid na
 				if (skill_produce_db[idx].itemlv > 10 && skill_produce_db[idx].itemlv <= 20 ) { //Cooking items.
 					clif_specialeffect(&sd->bl, EF_COOKING_FAIL, AREA);
 					if (sd->cook_mastery > 0)
```
Comment (Lemongrass3110) [inline src/map/skill.cpp:21990]:
```suggestion
```

File: src/map/skill.cpp [hunk 4]
```diff
@@ -21988,7 +21988,7 @@ bool skill_produce_mix(struct map_session_data *sd, uint16 skill_id, t_itemid na
...
-						pc_setglobalreg(sd, add_str(COOKMASTERY_VAR), sd->cook_mastery - ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) - ( ( ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) >> 1 ) * 3 ));
+						pc_setparam(sd, SP_COOKMASTERY, sd->cook_mastery - ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) - ( ( ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) >> 1 ) * 3 ));
```
Comment (Lemongrass3110) [inline src/map/skill.cpp:21991]:
```suggestion
					pc_setparam(sd, SP_COOKMASTERY, sd->cook_mastery - ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) - ( ( ( 1 << ((skill_produce_db[idx].itemlv - 11) / 2) ) >> 1 ) * 3 ));
```

**Review by Lemongrass3110 — APPROVED** [review]
Please remove the negative check in pc_setparam, then everything I suggested should be fine and logic (min/max values) is only in a single place!

**Comment by Lemongrass3110** [comment]
@Shaktohh can we get your input if it works as expected now?

**Comment by Shaktohh** [comment]
> @Shaktohh can we get your input if it works as expected now?

it works perfectly

**Pull request merged**